### PR TITLE
Complaint module form added with changes

### PR DESCRIPTION
**Pull request by sshivam95**

### CODE REVIEW

**Comment by pulsejet** [comment]
I've cleaned this up, but I can't say the design and colors are in line with the rest of the app ...

**Comment by pulsejet** [comment]
@sshivam95 make sure you're rebased on these changes if you make any more changes.

File: app/src/main/java/app/insti/fragment/FileComplaintFragment.java [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+        PlaceAutocompleteFragment fragment = (PlaceAutocompleteFragment) fragmentManager.findFragmentById(R.id.place_autocomplete_fragment);
+        android.app.FragmentTransaction ft = fragmentManager.beginTransaction();
+        ft.remove(fragment);
+        ft.commit();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        Toast.makeText(getContext(), getString(R.string.initial_message_file_complaint), Toast.LENGTH_SHORT).show();
```
Comment (pulsejet) [inline app/src/main/java/app/insti/fragment/FileComplaintFragment.java]: Don't show this here. It is
1) Detected as a screen overlay, so the user cannot allow on MM+
2) Plainly unnecessary

Comment (sshivam95) [reply app/src/main/java/app/insti/fragment/FileComplaintFragment.java]: This was told by ma'am in today's meeting to add a toast on start of this fragment

Comment (sshivam95) [reply app/src/main/java/app/insti/fragment/FileComplaintFragment.java]: Granted Permission for what?

Comment (pulsejet) [reply app/src/main/java/app/insti/fragment/FileComplaintFragment.java]: Map

Comment (sshivam95) [reply app/src/main/java/app/insti/fragment/FileComplaintFragment.java]: Ohk ok. Got it 👍

**Comment by ydidwania** [comment]
@sshivam95  I can see a lot of effort has already gone into this. Just want to point out a few design issues, which might later prevent this from merge. 

1.  The "Comments" and "Upvotes" should be icons like Reddit/Facebook app.
image
2. The color scheme isn't in line with the rest of the app. And the map should center on the location of the user.
image

**Comment by pulsejet** [comment]
Agree with @ydidwania. You could also fix the plus icon, which looks way out of place. Remember, for a new service, it's the design that matters the *most* of all other things. I'd also suggest increasing the padding around the cards and giving the reported badges round edges.

**Comment by sshivam95** [comment]
> @sshivam95 I can see a lot of effort has already gone into this. Just want to point out a few design issues, which might later prevent this from merge.
> 
> 1. The "Comments" and "Upvotes" should be icons like Reddit/Facebook app.
>    image
> 2. The color scheme isn't in line with the rest of the app. And the map should center on the location of the user.
>    image

@ydidwania Thank you for the feed back, Point 1 is noted and will be followed. As of point 2, there should be a button on the map which onClick will point to the user's current location. Believe me I have given a whole 2 week on the matter which is asked in point 2 and was unable to find a single solid solution.

**Comment by sshivam95** [comment]
> Agree with @ydidwania. You could also fix the plus icon, which looks way out of place. Remember, for a new service, it's the design that matters the _most_ of all other things. I'd also suggest increasing the padding around the cards and giving the reported badges round edges.

The plus icon is actually the zoom button which is a default from the google maps @pulsejet . @ydidwania , I can see the "No complaints at the moment" tag in the first picture. I am positive that this thing will only come when there is no complaint in the database. You might be using an older version of the repo. I'll make another commit tomorrow with all the changes @pulsejet @ydidwania

**Comment by pulsejet** [comment]
> The plus icon is actually the zoom button which is a default from the google maps @pulsejet 

I meant the green on grey plus icon beside "Add more Tags" @sshivam95

**Comment by sshivam95** [comment]
Oh ok sorry 😅

**Comment by sshivam95** [comment]
> > 1. The "Comments" and "Upvotes" should be icons like Reddit/Facebook app.
> >    image
@ydidwania . How did you added this compliant? May i know that. Because as far as I'm concerned, that ldap id is not with you.

**Comment by ydidwania** [comment]
> > > 1. The "Comments" and "Upvotes" should be icons like Reddit/Facebook app.
> > >    image
> > >    @ydidwania . How did you added this compliant? May i know that. Because as far as I'm concerned, that ldap id is not with you.

That was added @pulsejet

**Comment by pulsejet** [comment]
> @ydidwania . How did you added this compliant? May i know that. Because as far as I'm concerned, that ldap id is not with you.

I did that, with superuser access. Had added a few more with different LDAP ids and was deleting them when I realized we should have some screenshots. Who cleared the table before if not you @sshivam95 ?

**Comment by sshivam95** [comment]
I cleared the table after demonstrating the app to the professors on today's meeting. @pulsejet. But I am not able to understand that why the tag " No complaints at the moment" is coming when I designed it to be gone. Also if you want to test the adding of complaints, try using the form because it's result is different as to what if added from the database

**Comment by pulsejet** [comment]
> I am not able to understand that why the tag " No complaints at the moment" is coming

It's possible that the branch I built wasn't updated with the changes you did later.

> Also if you want to test the adding of complaints, try using the form because it's result is different as to what if added from the database

I had to test for different users.

**Comment by sshivam95** [comment]
>I had to test for different users.

Oh ok. But just try with your's id and see the difference

**Comment by sshivam95** [comment]
@pulsejet @ydidwania . I have acknowledged and made the changes. Please review and give feedbacks. Thank you

**Comment by sshivam95** [comment]
> Haven't had time to check the design changes yet, but the PR itself looks clean enough now. A couple of comments that come to mind:
> 
> 1. At some point, the list of complaints will have to be lazy loaded similar to news and blog, since we don't want to load hundreds at once. This should be trivial with the reusable recycler view fragment.
> 2. There's already a placeholder image for user in resources matching the theme, so the current could be refactored to use the same if not already done.
> 3. Adding loading indicator and transition animations to fragments wherever not done yet.

Acknowledged. A few doubts though, in point 1, the lazy loading that you are talking about, will be from the backend side right?
The image placeholder, in point 2, is been used at all points necessary. I'll look into the point 3.
Thanks

**Comment by sshivam95** [comment]
One request @pulsejet . Please let me know if this PR is ready, then I'll add the comments to make the code more understandable for the next person after me.

**Comment by pulsejet** [comment]
> the lazy loading that you are talking about, will be from the backend side right?

Both, backend and the app. The backend part should be trivial with the existing framework for news and blogs.

> Please let me know if this PR is ready, then I'll add the comments to make the code more understandable for the next person after me.

Now looks like a good time to do this.

**Comment by pulsejet** [comment]
@sshivam95 please also check the review by @codacy-bot

**Pull request merged**